protocols/relay/src/priv_client.rs

```diff
@@ -391,7 +391,8 @@ pub struct Connection {
 }
 
 /// A [`NegotiatedSubstream`] acting as a [`ConnectionState`].
-pub enum ConnectionState {
+#[allow(dead_code)]
```

Comment:
What is dead code here? We should fix that if it is completely internal.

Reply:
@thomaseizinger Related to this comment: https://github.com/libp2p/rust-libp2p/pull/3829#discussion_r1176820157. Maybe it's a mistake from my compiler, just pushed now without this to see if the clippy error appears here.

Reply:
I fixed it and added a comment :)